gradle-jdks/src/main/java/com/palantir/gradle/jdks/JdksPlugin.java

```diff
@@ -90,7 +90,15 @@ private JavaInstallationMetadata javaInstallationForLanguageVersion(
             JdkManager jdkManager,
             JavaLanguageVersion javaLanguageVersion) {
 
-        String version = jdkExtension.getJdkVersion().get();
+        Os currentOs = CurrentOs.current();
+        Arch currentArch = CurrentArch.current();
+
+        String version = jdkExtension
+                .jdkFor(currentOs)
+                .jdkFor(currentArch)
+                .getJdkVersion()
+                .get();
```

Comment:
curious, why force property retrieval here? do you not get into races with configuration?

Reply:
This method, `javaInstallationForLanguageVersion` is called from a lazy location, but you're right this could probably be even more lazy to avoid problems.

Reply:
ah